dmutils/content_loader.py

```diff
@@ -111,6 +114,19 @@ def get_question(self, question_id):
             if question:
                 return question
 
+    def get_question_number(self, question_id):
+        index = 0
+        for section in self.sections:
+            for question in section.questions:
+                index += 1
+                if question['id'] == question_id:
```

Comment:
This is a lot of looping, for every question we loop through every question. If this is just being done in the constructor we could do this with one pass.